src/aspire/estimation/covar2d.py

```diff
@@ -101,29 +101,31 @@ def get_mean(self, coeffs, ctf_fb=None, ctf_idx=None):
         :param coeffs: A coefficient vector (or an array of coefficient vectors) to be averaged.
         :param ctf_fb: The CFT functions in the FB expansion.
         :param ctf_idx: An array of the CFT function indices for all 2D images.
+            If ctf_fb or ctf_idx is None, the identity filter will be applied.
         :return: The mean value vector for all images.
         """
         if coeffs.size == 0:
             raise RuntimeError('The coefficients need to be calculated!')
 
         if (ctf_fb is None) or (ctf_idx is None):
-            return self._get_mean(coeffs)
+            ctf_idx = np.zeros(coeffs.shape[1], dtype=int)
+            ctf_fb = [blk_diag_eye(blk_diag_partition(RadialCTFFilter().fb_mat(self.basis)))]
 
-        b = np.zeros((self.basis.count, 1), dtype=self.as_type)
+        b = np.zeros((self.basis.count, 1), dtype=coeffs.dtype)
```

Comment:
Again, just `np.zeros(self.basis.count, dtype=coeffs.dtype`.

Reply:
I have to reshape it back to two dimension array later due to the function of `blk_diag_apply`. I will keep it unchanged now. Open the issue #111 for reimplement `blk_diag_apply`  and other functions.

Reply:
> I have to reshape it back to two dimension array later due to the function of blk_diag_apply. 

In that case, it's preferable to do the reshaping when calling `blk_diag_apply` instead of dragging these 2D arrays around when we should have vectors.

Reply:
Done.